lib/file_transfer_agent/encrypt_util.js

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2015-2021 Snowflake Computing Inc. All rights reserved.
+ */
+
+var path = require('path');
+var tmp = require('tmp');
+
+const AES_CBC = 'aes-128-cbc';
+const AES_ECB = 'aes-128-ecb';
+const AES_BLOCK_SIZE = 128;
+const blockSize = parseInt(AES_BLOCK_SIZE / 8);  // in bytes
+
+// Material Descriptor
+function MaterialDescriptor(smkId, queryId, keySize)
+{
+  return {
+    "smkId": smkId,
+    "queryId": queryId,
+    "keySize": keySize
+  }
+}
+
+// Encryption Material
+function EncryptionMetadata(key, iv, matDesc)
+{
+  return {
+    "key": key,
+    "iv": iv,
+    "matDesc": matDesc
+  }
+}
+
+exports.EncryptionMetadata = EncryptionMetadata;
+
+/**
+ * Creates an encryption utility object.
+ *
+ * @param {module} encrypt
+ * @param {module} filestream
+ * @param {module} temp
+ * 
+ * @returns {Object}
+ * @constructor
+ */
+function encrypt_util(encrypt, filestream, temp)
+{
+  const crypto = typeof encrypt !== "undefined" ? encrypt : require('crypto');
+  const fs = typeof filestream !== "undefined" ? filestream : require('fs');
+  const tmp = typeof temp !== "undefined" ? temp : require('tmp');
+
+  /**
+   * Generate a buffer with random bytes given a size.
+   *
+   * @param {Number} byteLength
+   *
+   * @returns {Buffer} of size byteLength
+   */
+  function getSecureRandom(byteLength)
+  {
+    return crypto.randomBytes(byteLength);
+  }
+
+  /**
+  * Convert a material descriptor object's values to unicode.
+  *
+  * @param {Object} matDesc
+  *
+  * @returns {Object}
+  */
+  function matDescToUnicode(matDesc)
+  {
+    matDesc['smkId'] = matDesc['smkId'].toString();
+    matDesc['keySize'] = matDesc['keySize'].toString();
+    var newMatDesc = JSON.stringify(matDesc);
+    return newMatDesc;
+  }
+
+  /**
+  * Encrypt file using AES algorithm.
+  *
+  * @param {Object} encryptionMaterial
+  * @param {String} inFileName
+  * @param {String} tmpDir
+  * @param {Number} chunkSize
+  *
+  * @returns {Object}
+  */
+  this.encryptFile = async function (encryptionMaterial, inFileName,
+    tmpDir = null, chunkSize = blockSize * 4 * 1024)
+  {
+    // Get decoded key from base64 encoded value
+    var decodedKey = Buffer.from(encryptionMaterial['queryStageMasterKey'], 'base64');
```

Comment:
nit: Turn 'queryStageMasterKey' into a named constant

Reply:
Same with 'base64' since its used in a few places

Reply:
Done